src/preload.ts

```diff
@@ -1,17 +1,19 @@
-const { setTitle } = require("./scripts/window-functions");
-const { dialog, process, Notification } = require("@electron/remote");
-const { store, settings } = require("./scripts/settings");
-const { ipcRenderer } = require("electron");
-const { app } = require("@electron/remote");
-const { downloadFile } = require("./scripts/download");
-const statuses = require("./constants/statuses");
-const hotkeys = require("./scripts/hotkeys");
-const globalEvents = require("./constants/globalEvents");
-const { skipArtists, updateFrequency, customCSS } = require("./constants/settings");
+import { Notification, app, dialog } from "@electron/remote";
+import { ipcRenderer } from "electron";
+import Player from "mpris-service";
+import { globalEvents } from "./constants/globalEvents";
+import { settings } from "./constants/settings";
+import { statuses } from "./constants/statuses";
+import { Options } from "./models/options";
+import { downloadFile } from "./scripts/download";
+import { addHotkey } from "./scripts/hotkeys";
+
+import { settingsStore } from "./scripts/settings";
+import { setTitle } from "./scripts/window-functions";
 const notificationPath = `${app.getPath("userData")}/notification.jpg`;
 const appName = "Tidal Hifi";
 let currentSong = "";
-let player;
+let player: any;
```

Comment:
Did you ever think about writing your own declaration for this?

Reply:
Yep. I first transferred everything to any (or simple types) and then when it worked with the compile step in place I went back and removed all the any's (or so I thought).

Good catch, I'll write a declaration for it!

Did you test this branch by chance?
Seems to work for me 

PS: the "relaunch" button is broken again, haven't yet verified whether that was electron or TS... probably electron.

Reply:
Yeah, experiencing the same issue with the "relaunch" button I highly doubt it has something to do with TS as it used to be broken already since it always only exited the app without restarting since electron updates were done a while back on this project.

Reply:
Yeah.. my thoughts exactly. I'll write up type declarations later and just merge this.

Thanks :)